### DISCUSSION TITLE
How to allow rewrites when using static export on local development?

### DISCUSSION THREAD

**Discussion by SanderPeeters** [vercel/next.js#48567] [Help] [answered]
### Summary

I'm working on a Next app and I like to use the static export. I'm using react-router for my client routing. In my public folder I've added a `staticwebapp.config.js` for Azure to make sure every route is rewritten to the built index.html and this works fine as you can see here: https://delightful-hill-09f273903.3.azurestaticapps.net/

When I'm working locally on my computer and serve my app using `next dev` the client routing works, but when I refresh I get to see the 404 page. Which makes sense because I have no rewrites configured for local development. The docs are telling me that configuring `rewrites` is not supported when using static export. But I was wondering how I should enable this or override this for local development? 

https://beta.nextjs.org/docs/configuring/static-export#unsupported-features

Or am I missing something here?

### Additional information

```js
next.config.js

/** @type {import('next').NextConfig} */
module.exports = {
    output: 'export',
    experimental: {
        appDir: true,
    },
}
```
staticwebapp.config.js
```
{
    "navigationFallback": {
        "rewrite": "index.html",
        "exclude": ["/static/media/*.{png,jpg,jpeg,gif,bmp}", "/static/css/*"]
    },
    "mimeTypes": {
        ".json": "text/json"
    }
}
```
```


### Example

_No response_

**Comment by icyJoseph** [comment, marked as answer]
I guess the root issue is the usage of use react-router, you are just like, hacking it, and the dev server not working is but a symptom of the hack....

That being said, I can help you, I think... but please try to move away from  react-router.

The good news is that the config file can be a function. 

```js
const { PHASE_DEVELOPMENT_SERVER } = require("next/constants");

/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
};

module.exports = (phase) => {
  if (phase === PHASE_DEVELOPMENT_SERVER) {
    /** @type {import('next').NextConfig} */
    return {
      ...nextConfig,
      async rewrites() {
        return {
          beforeFiles: [
            {
              source: "/foo/bar",
              destination: "/",
            },
          ],
        };
      },
    };
  }

  return nextConfig;
};

```

- https://nextjs.org/docs/api-reference/next.config.js/introduction

And that should do the trick, I think. Because during build time, and export time, the phase is:

- 'phase-production-build'
- 'phase-export'

**Reply by howard-os** [reply]
I had to use this instead, otherwise I got console errors as it tried to parse some html output as scripts (unexpected <)

```
return [
    {
        source: '/:any*',
        destination: '/'
    }
];
```
